src/v/cluster/metrics_reporter.cc

```diff
@@ -101,6 +105,53 @@ address parse_url(const ss::sstring& url) {
     }
     return ret;
 }
+
+ss::sstring get_hostname() {
+    std::array<char, HOST_NAME_MAX> hostname{};
+    if (::gethostname(hostname.data(), hostname.size()) != 0) {
+        return {};
+    }
+
+    return hostname.data();
+}
+
+ss::sstring get_domainname() {
+    std::array<char, HOST_NAME_MAX> domainname{};
+    if (::getdomainname(domainname.data(), domainname.size()) != 0) {
+        return {};
+    }
+
+    return domainname.data();
```

Comment:
Isn't this an issue if `getdomainname` returns HOST_NAME_MAX bytes, the `null` terminator isn't written then right?

```
getdomainname() returns the null-terminated domain name in the
       character array name, which has a length of len bytes.  If the
       null-terminated domain name requires more than len bytes,
       getdomainname() returns the first len bytes (glibc) or gives an
       error (libc).
```

Should we request HOST_NAME_MAX - 1 or have the array have +1 chars for a null byte?

Reply:
Hmm, but later on in the man page it states that:

```
Since Linux 1.0, the limit on the length of a domain name,
       including the terminating null byte, is 64 bytes.  In older
       kernels, it was 8 bytes.
```